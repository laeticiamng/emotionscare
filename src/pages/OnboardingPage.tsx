import React, { useState, useEffect } from 'react';
import { useNavigate, useSearchParams } from 'react-router-dom';
import { useAuth } from '@/contexts/AuthContext';
import { Button } from '@/components/ui/button';
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '@/components/ui/card';
import { Input } from '@/components/ui/input';
import { Label } from '@/components/ui/label';
import { RadioGroup, RadioGroupItem } from '@/components/ui/radio-group';
import { Switch } from '@/components/ui/switch';
import { FontSize, ThemeName, User } from '@/types';
import { useUserMode } from '@/contexts/UserModeContext';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';

const OnboardingPage = () => {
  const { user, updateUser } = useAuth();
  const { userMode } = useUserMode();
  const navigate = useNavigate();
  const [searchParams] = useSearchParams();
  const mode = searchParams.get('mode') || (userMode === 'b2c' ? 'personal' : 'business');
  
  const [name, setName] = useState<string>(user?.name || '');
  const [theme, setTheme] = useState<ThemeName>('light');
  const [fontSize, setFontSize] = useState<FontSize>('medium');
  const [notificationsEnabled, setNotificationsEnabled] = useState<boolean>(true);
  const [emailNotifications, setEmailNotifications] = useState<boolean>(true);
  const [department, setDepartment] = useState<string>('');
  const [position, setPosition] = useState<string>('');
  const [isLoading, setIsLoading] = useState<boolean>(false);
  
  useEffect(() => {
    // If user has already completed onboarding, redirect to dashboard
    if (user?.onboarded) {
      navigate('/dashboard');
    }
  }, [user, navigate]);
  
  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    setIsLoading(true);
    
    try {
      if (!user) return;
      
      // Common user preferences
      const updatedUser: User = {
        ...user,
        name,
        preferences: {
          theme,
          fontSize,
          fontFamily: 'inter',
          notifications_enabled: notificationsEnabled,
          email_notifications: emailNotifications,
          push_notifications: true
        },
        onboarded: true,
        joined_at: new Date().toISOString(),
        avatar_url: user.avatar_url || ''
      };
      
      // Add business-specific fields if in business mode
      if (mode === 'business' || userMode === 'b2b-collaborator' || userMode === 'b2b-admin') {
        updatedUser.department = department;
        updatedUser.position = position;
      }
      
      await updateUser(updatedUser);
      navigate('/dashboard');
    } catch (error) {
      console.error('Error during onboarding:', error);
    } finally {
      setIsLoading(false);
    }
  };
  
  return (
    <div className="flex items-center justify-center min-h-screen bg-muted/20 p-4">
      <Card className="w-full max-w-md">
        <CardHeader className="text-center">
          <CardTitle className="text-2xl">Bienvenue !</CardTitle>
          <CardDescription>
            {mode === 'business' 
              ? 'Configurez votre profil professionnel' 
              : 'Configurez votre expérience pour commencer'}
          </CardDescription>
        </CardHeader>
        <CardContent>
          <form onSubmit={handleSubmit} className="space-y-6">
            <div className="space-y-2">
              <Label htmlFor="name">Votre nom</Label>
              <Input 
                id="name"
                value={name}
                onChange={(e) => setName(e.target.value)}
                placeholder="Comment souhaitez-vous être appelé ?"
                required
              />
            </div>
            
            {/* Business-specific fields */}
            {(mode === 'business' || userMode === 'b2b-collaborator' || userMode === 'b2b-admin') && (
              <>
                <div className="space-y-2">
                  <Label htmlFor="department">Département</Label>
                  <Select value={department} onValueChange={setDepartment}>
                    <SelectTrigger id="department">
                      <SelectValue placeholder="Sélectionnez votre département" />
                    </SelectTrigger>
                    <SelectContent>
                      <SelectItem value="rh">Ressources Humaines</SelectItem>
                      <SelectItem value="marketing">Marketing</SelectItem>
                      <SelectItem value="tech">Technique / IT</SelectItem>
                      <SelectItem value="finance">Finance</SelectItem>
                      <SelectItem value="operations">Opérations</SelectItem>
                      <SelectItem value="direction">Direction</SelectItem>
                      <SelectItem value="other">Autre</SelectItem>
                    </SelectContent>
                  </Select>
                </div>
                
                <div className="space-y-2">
                  <Label htmlFor="position">Poste</Label>
                  <Input 
                    id="position"
                    value={position}
                    onChange={(e) => setPosition(e.target.value)}
                    placeholder="Votre fonction dans l'entreprise"
                  />
                </div>
              </>
            )}
            
            <div className="space-y-2">
              <Label>Thème</Label>
              <RadioGroup 
                value={theme} 
                onValueChange={(value) => setTheme(value as ThemeName)}
                className="flex space-x-2"
              >
                <div className="flex items-center space-x-2">
                  <RadioGroupItem value="light" id="light" />
                  <Label htmlFor="light">Clair</Label>
                </div>
                <div className="flex items-center space-x-2">
                  <RadioGroupItem value="dark" id="dark" />
                  <Label htmlFor="dark">Sombre</Label>
                </div>
                <div className="flex items-center space-x-2">
                  <RadioGroupItem value="system" id="system" />
                  <Label htmlFor="system">Système</Label>
                </div>
              </RadioGroup>
            </div>
            
            <div className="space-y-2">
              <Label>Taille de police</Label>
              <RadioGroup 
                value={fontSize} 
                onValueChange={(value) => setFontSize(value as FontSize)}
                className="flex space-x-2"
              >
                <div className="flex items-center space-x-2">
                  <RadioGroupItem value="small" id="small" />
                  <Label htmlFor="small">Petite</Label>
                </div>
                <div className="flex items-center space-x-2">
                  <RadioGroupItem value="medium" id="medium" />
                  <Label htmlFor="medium">Moyenne</Label>
                </div>
                <div className="flex items-center space-x-2">
                  <RadioGroupItem value="large" id="large" />
                  <Label htmlFor="large">Grande</Label>
                </div>
              </RadioGroup>
            </div>
            
            <div className="space-y-4">
              <Label>Notifications</Label>
              <div className="flex items-center justify-between">
                <Label htmlFor="notifications" className="text-sm">
                  Notifications dans l'application
                </Label>
                <Switch 
                  id="notifications"
                  checked={notificationsEnabled}
                  onCheckedChange={setNotificationsEnabled}
                />
              </div>
              
              <div className="flex items-center justify-between">
                <Label htmlFor="email-notifications" className="text-sm">
                  Notifications par email
                </Label>
                <Switch 
                  id="email-notifications"
                  checked={emailNotifications}
                  onCheckedChange={setEmailNotifications}
                />
              </div>
            </div>
            
            <Button type="submit" className="w-full" disabled={isLoading}>
              {isLoading ? "Chargement..." : "Commencer"}
            </Button>
          </form>
        </CardContent>
      </Card>
    </div>
  );
};

export default OnboardingPage;
