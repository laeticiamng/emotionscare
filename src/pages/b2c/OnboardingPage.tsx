
import React from 'react';
import { motion } from 'framer-motion';
import { Heart, ArrowLeft } from 'lucide-react';
import { Button } from '@/components/ui/button';
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '@/components/ui/card';
import { useNavigate } from 'react-router-dom';

const OnboardingPage: React.FC = () => {
  const navigate = useNavigate();

  return (
    <div className="min-h-screen bg-gradient-to-br from-green-50 to-emerald-50 dark:from-gray-900 dark:to-gray-800 flex items-center justify-center p-4">
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ duration: 0.6 }}
        className="w-full max-w-2xl"
      >
        <Button
          onClick={() => navigate('/b2c/dashboard')}
          variant="ghost"
          className="mb-6"
        >
          <ArrowLeft className="mr-2 h-4 w-4" />
          Aller au tableau de bord
        </Button>

        <Card>
          <CardHeader className="text-center">
            <div className="flex justify-center mb-4">
              <Heart className="h-12 w-12 text-green-500" />
            </div>
            <CardTitle className="text-2xl">Bienvenue dans EmotionsCare</CardTitle>
            <CardDescription>
              Commençons votre parcours de bien-être émotionnel
            </CardDescription>
          </CardHeader>
          <CardContent className="p-8">
            <div className="space-y-6">
              <div className="text-center">
                <h3 className="text-lg font-semibold mb-4">Configuration de votre profil</h3>
                <p className="text-gray-600 dark:text-gray-300 mb-6">
                  Nous allons personnaliser votre expérience pour mieux vous accompagner
                </p>
              </div>
              
              <div className="space-y-4">
                <div className="p-4 bg-green-50 dark:bg-green-900/20 rounded-lg">
                  <h4 className="font-medium text-green-800 dark:text-green-300">Étape 1</h4>
                  <p className="text-sm text-green-600 dark:text-green-400">Définir vos objectifs de bien-être</p>
                </div>
                <div className="p-4 bg-blue-50 dark:bg-blue-900/20 rounded-lg">
                  <h4 className="font-medium text-blue-800 dark:text-blue-300">Étape 2</h4>
                  <p className="text-sm text-blue-600 dark:text-blue-400">Configurer vos préférences</p>
                </div>
                <div className="p-4 bg-purple-50 dark:bg-purple-900/20 rounded-lg">
                  <h4 className="font-medium text-purple-800 dark:text-purple-300">Étape 3</h4>
                  <p className="text-sm text-purple-600 dark:text-purple-400">Découvrir les fonctionnalités</p>
                </div>
              </div>

              <Button 
                onClick={() => navigate('/b2c/dashboard')}
                className="w-full bg-green-500 hover:bg-green-600"
              >
                Commencer l'aventure
              </Button>
            </div>
          </CardContent>
        </Card>
      </motion.div>
    </div>
  );
};

export default OnboardingPage;
