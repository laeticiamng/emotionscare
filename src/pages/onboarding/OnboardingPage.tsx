import React, { useEffect } from 'react';
import { OnboardingStepper } from '@/components/onboarding/OnboardingStepper';
import { StepWelcome } from '@/components/onboarding/StepWelcome';
import { StepProfile } from '@/components/onboarding/StepProfile';
import { StepGoals } from '@/components/onboarding/StepGoals';
import { useOnboarding } from '@/hooks/useOnboarding';
import { useRouter } from '@/hooks/useRouter';

// Lazy load steps to improve initial load time
const StepSensors = React.lazy(() => import('@/components/onboarding/StepSensors'));
const StepNotifications = React.lazy(() => import('@/components/onboarding/StepNotifications'));
const StepSummary = React.lazy(() => import('@/components/onboarding/StepSummary'));

const OnboardingPage: React.FC = () => {
  const router = useRouter();
  const {
    step,
    completed,
    loading,
    error,
    profileDraft,
    goalsDraft,
    sensorsDraft,
    start,
    saveProfile,
    saveGoals,
    saveSensors,
    enableNotifications,
    complete,
    skip,
    next,
    prev,
  } = useOnboarding();

  // Redirect if already completed
  useEffect(() => {
    if (completed) {
      router.push('/dashboard');
    }
  }, [completed, router]);

  // Handle skip to dashboard
  const handleSkip = () => {
    skip();
    router.push('/dashboard');
  };

  // Handle completion and redirect
  const handleComplete = async () => {
    const success = await complete();
    if (success) {
      router.push('/dashboard');
    }
  };

  // Handle profile save and advance
  const handleProfileSubmit = async (profile: typeof profileDraft) => {
    const success = await saveProfile(profile);
    if (success) {
      next();
    }
    return success;
  };

  // Handle goals save and advance
  const handleGoalsSubmit = async (goals: typeof goalsDraft) => {
    const success = await saveGoals(goals);
    if (success) {
      next();
    }
    return success;
  };

  // Handle sensors save and advance
  const handleSensorsSubmit = async (sensors: typeof sensorsDraft) => {
    const success = await saveSensors(sensors);
    if (success) {
      next();
    }
    return success;
  };

  if (loading && step === 0) {
    return (
      <div className="min-h-screen flex items-center justify-center">
        <div className="text-center">
          <div className="animate-spin w-8 h-8 border-4 border-primary border-t-transparent rounded-full mx-auto mb-4" />
          <p className="text-muted-foreground">Initialisation...</p>
        </div>
      </div>
    );
  }

  return (
    <main className="min-h-screen bg-gradient-to-b from-background to-muted p-6">
      <div className="container mx-auto max-w-4xl">
        {/* Show stepper except for welcome step */}
        {step > 0 && <OnboardingStepper currentStep={step} />}

        {/* Error display */}
        {error && (
          <div className="mb-6 p-4 bg-destructive/10 border border-destructive/20 rounded-lg text-center">
            <p className="text-destructive text-sm">{error}</p>
          </div>
        )}

        {/* Step content */}
        <div className="mb-8">
          {step === 0 && (
            <StepWelcome 
              onNext={step === 0 ? () => { start(); next(); } : next}
              onSkip={handleSkip}
            />
          )}

          {step === 1 && (
            <StepProfile
              onSubmit={handleProfileSubmit}
              onBack={prev}
              initialData={profileDraft}
            />
          )}

          {step === 2 && (
            <StepGoals
              onSubmit={handleGoalsSubmit}
              onBack={prev}
              initialData={goalsDraft}
            />
          )}

          {step === 3 && (
            <React.Suspense 
              fallback={
                <div className="flex justify-center py-8">
                  <div className="animate-spin w-6 h-6 border-4 border-primary border-t-transparent rounded-full" />
                </div>
              }
            >
              <StepSensors
                onSubmit={handleSensorsSubmit}
                onBack={prev}
                initialData={sensorsDraft}
              />
            </React.Suspense>
          )}

          {step === 4 && (
            <React.Suspense 
              fallback={
                <div className="flex justify-center py-8">
                  <div className="animate-spin w-6 h-6 border-4 border-primary border-t-transparent rounded-full" />
                </div>
              }
            >
              <StepNotifications
                onNext={next}
                onBack={prev}
                onEnable={enableNotifications}
              />
            </React.Suspense>
          )}

          {step === 5 && (
            <React.Suspense 
              fallback={
                <div className="flex justify-center py-8">
                  <div className="animate-spin w-6 h-6 border-4 border-primary border-t-transparent rounded-full" />
                </div>
              }
            >
              <StepSummary
                onFinish={handleComplete}
                onBack={prev}
              />
            </React.Suspense>
          )}
        </div>

        {/* Skip option (always available) */}
        {step > 0 && step < 5 && (
          <div className="text-center">
            <button
              onClick={handleSkip}
              className="text-sm text-muted-foreground hover:text-foreground transition-colors"
            >
              Passer l'introduction et aller au tableau de bord
            </button>
          </div>
        )}
      </div>
    </main>
  );
};

export default OnboardingPage;