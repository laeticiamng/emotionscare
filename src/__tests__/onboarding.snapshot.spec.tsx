import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import OnboardingPage from "@/app/onboarding/page";

describe("OnboardingPage", () => {
  it("rend l’étape 1 et matche le snapshot", () => {
    const { container, getByText } = render(<OnboardingPage />);
    expect(getByText(/Bienvenue/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
