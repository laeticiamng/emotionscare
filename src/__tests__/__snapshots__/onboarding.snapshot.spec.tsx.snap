// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OnboardingPage > rend lâ€™Ã©tape 1 et matche le snapshot 1`] = `
<div>
  <main aria-label="Parcours dâ€™onboarding">
    <div>
      Bienvenue
    </div>
    <div>
      <section aria-labelledby="step1">
        <h2 id="step1">Bienvenue ğŸ‘‹</h2>
        <p>On va rÃ©gler quelques prÃ©fÃ©rences pour personnaliser ton expÃ©rience.</p>
        <button data-ui="primary-cta">Commencer</button>
      </section>
    </div>
  </main>
</div>
`;
